sql/src/test/java/com/amazon/opendistroforelasticsearch/sql/sql/antlr/SQLSyntaxParserTest.java

```diff
@@ -29,13 +28,36 @@
 
   @Test
   public void canParseSelectLiterals() {
-    ParseTree parseTree = parser.parse("SELECT 123, 'hello'");
-    assertNotNull(parseTree);
+    assertNotNull(parser.parse("SELECT 123, 'hello'"));
+  }
+
+  @Test
+  public void canParseIndexNameWithDate() {
+    assertNotNull(parser.parse("SELECT * FROM logs_2020_01"));
+    assertNotNull(parser.parse("SELECT * FROM logs-2020-01"));
+  }
+
+  @Test
+  public void canParseHiddenIndexName() {
+    assertNotNull(parser.parse("SELECT * FROM .kibana"));
+  }
+
+  @Test
+  public void canNotParseIndexNameWithSpecialChar() {
+    assertThrows(SyntaxCheckException.class,
+        () -> parser.parse("SELECT * FROM hello+world"));
+  }
+
+  @Test
+  public void canParseIndexNameWithSpecialCharQuoted() {
+    assertNotNull(parser.parse("SELECT * FROM `hello+world`"));
+    assertNotNull(parser.parse("SELECT * FROM \"hello$world\""));
```

Comment:
could you add test with single quota identifier, e.g. SELECT * FROM 'hello+world'

Reply:
I missed this. Will add test case and change grammar. Thanks!